src/client.rs

```diff
@@ -838,6 +844,48 @@ impl Client {
         .await;
     }
 
+    async fn on_stats(&self, message: Message) {
+        if !*self.registered.lock().await {
+            return;
+        }
```

Comment:
This is unnecessary, we now check the registered flag in the on_message function.

Reply:
ok master